R-package/tests/testthat/test_Predictor.R

```diff
@@ -68,6 +68,7 @@ test_that("start_iteration works correctly", {
     bst <- lightgbm(
         data = as.matrix(train$data)
         , label = train$label
+        , nthreads = 1L
```

Comment:
This change and similar changes I see in this PR (like in the vignettes) where `nthreads = 1L` is being added don't seem necessary to implement support for passing through `num_threads` as a top-level parameter in `lightgbm::lightgbm()`.

Please explain why these are necessary or remove them.

Reply:
They are necessary, because CRAN policies say that the examples, tests, and vignettes should only use 1 or 2 threads. Otherwise, there would have to be a version for running on them on CRAN and another version for running it locally, or it'd have to run a purrr wrapper or similar that detects whether it's being run on CRAN servers or not.

Reply:
Can we then set this param to `2L` with the aim to slightly speed up our CI tests?
Also, I think it would be nice to have `NTHREADS` global variable similarly to the `VERBOSITY` one
https://github.com/microsoft/LightGBM/blob/820ae7e65e90860061ec89cc4a1ad0a39b96e051/R-package/tests/testthat/test_lgb.Booster.R#L1-L3

Reply:
At https://cran.r-project.org/web/packages/policies.html, I see the following

> If running a package uses multiple threads/cores it must never use more than two simultaneously: the check farm is a shared resource and will typically be running many checks simultaneously.

Thanks for bringing it to our attention. If you're able to share links to specific documentation for such claims in the future, it would save maintainers some time and be really appreciated.

I'm open to limiting all of the examples / tests to 2 threads, but don't believe it needs to be done in this PR. I've documented the work to limit the number of threads used in #5102, and will open a separate PR doing that in a way that works with LightGBM's CI process.

Please remove these changes where you are passing `num_threads = 1L` from this PR, and limit this PR to only exposing `num_threads` as a keyword argument to `lightgbm::lightgbm()`.